pallets/parachain-staking/src/lib.rs

```diff
@@ -1409,14 +1409,18 @@ pub mod pallet {
 			);
 			ensure_signed(origin)?;
 
-			let collators = <SelectedCandidates<T>>::get();
+			let mut collators_len = 0usize;
 			let max_collators = <TotalSelected<T>>::get();
 
+			if let Some(len) = <SelectedCandidates<T>>::decode_len() {
+				collators_len = len;
+			};
+
```

Comment:
I would be paranoid about the way you handle this, maybe initialize `collators_len` to `T::MaxCandidates::get()` so that it falls back to the most expensive case.

IIRC, in practice, `decode_len()` only returns `None` if there is a problem reading storage (e.g. corruption). But always good to be defensive.

Reply:
I wouldn't initialize `collators_len` to `T::MaxCandidates::get()`, just because in that case the statement of `collators_len * 3 > (max_collators * 2) as usize` would be true within an unreal scenario. This condition is important as it tells us if we are going under the 66% of `TotalSelected` to prevent any stall of the chain.

The reason to initialize it to zero is because if decode_len() returns None, the extrinsic will simply fail and return `TooLowCollatorCountToNotifyAsInactive` error.

Reply:
Ah, good points, I think it's ok as written then.